src/System.Management.Automation/utils/tracing/PSSysLogProvider.cs

```diff
@@ -102,6 +102,22 @@ internal override void LogAmsiUtilStateEvent(string state, string context)
             WriteEvent(PSEventId.Amsi_Init, PSChannel.Analytic, PSOpcode.Method, PSLevel.Informational, PSTask.Amsi, (PSKeyword)0x0, state, context);
         }
 
+        /// <summary>
+        /// Provider interface function for logging WDAC query event.
+        /// </summary>
+        /// <param name="queryName">Name of the WDAC query.</param>
+        /// <param name="fileName">Name of script file for policy query. Can be null value.</param>
+        /// <param name="querySuccess">Query call succeed code.</param>
+        /// <param name="queryResult">Result code of WDAC query.</param>
+        internal override void LogWDACQueryEvent(
+            string queryName,
+            string fileName,
+            int querySuccess,
+            int queryResult)
+        {
+            WriteEvent(PSEventId.WDAC_Query, PSChannel.Analytic, PSOpcode.Method, PSLevel.Informational, PSTask.WDAC, (PSKeyword)0x0, queryName, fileName, querySuccess, queryResult);
```

Comment:
This is a code path that shouldn't be reached.  Should be just have a debug assert that we shouldn't get here?

Reply:
I don't think an assert is needed here.  I see that AMSI also has an implementation but also should never be hit.  But in the future if we extend to non-Windows platforms then the logging should just work.